packages/manager/modules/billing-components/src/components/commitment-duration/CommitmentDuration.class.js

```diff
@@ -0,0 +1,31 @@
+import { maxBy } from 'lodash-es';
+
+export default class CommitmentDuration {
+  /**
+   *
+   * @param {number} duration commitment duration in months
+   * @param {Commitment[]} commitments commitments available for the given duration
+   * @param {Pricing} currentPricing current pricing for the service
+   */
+  constructor(duration, commitments, currentPricing) {
+    Object.assign(this, {
+      duration,
+      commitment: maxBy(commitments, 'pricing.monthlyPriceValue'),
+    });
+
+    if (currentPricing) {
+      this.savings = currentPricing.getDiff(
+        this.commitment.pricing,
+        this.duration,
+      );
+    }
+  }
+
+  get monthlyDuration() {
+    return moment.duration(this.duration).asMonths();
+  }
+
+  get monthlyPrice() {
+    return this.commitment.pricing.monthlyPrice.text;
```

Comment:
If we doubt its availability
```suggestion
    return this.commitment.pricing?.monthlyPrice?.text;
```

Reply:
In fact this should have been a `git mv`